server/chat.ts

```diff
@@ -758,14 +758,16 @@ export class CommandContext extends MessageContext {
 	shouldBroadcast() {
 		return this.cmdToken === BROADCAST_TOKEN;
 	}
-	canBroadcast(ignoreCooldown?: boolean, suppressMessage?: string | null) {
+	canBroadcast(ignoreCooldown?: boolean, suppressMessage?: string | null, image?: boolean) {
 		if (!this.broadcasting && this.shouldBroadcast()) {
 			if (this.room instanceof Rooms.GlobalRoom) {
 				this.errorReply(`You have no one to broadcast this to.`);
 				this.errorReply(`To see it for yourself, use: /${this.message.slice(1)}`);
 				return false;
 			}
-			if (!this.pmTarget && !this.user.can('broadcast', null, this.room)) {
+			if (
+				!this.pmTarget && (image ? !this.user.can('showimage', null, this.room) : !this.user.can('broadcast', null, this.room))
+			) {
```

Comment:
People should not be able to use `!show` in PMs without auth... We definitely don't want it getting used for porn and shock images.

Reply:
It should probably also not be allowed in groupchats.

Reply:
I've disabled it in temporary rooms with this commit.
It however is not usable in pms.

Reply:
It's not usable in PMs because you don't have the `"!show": true` sigil, but it's allowed in `canBroadcast`. I think that should be fixed, because the code is confusing about that point currently.

Reply:
Should gvoices+ be able to use it in PMs?

Reply:
Yeah.